libreoffice-core/sw/source/filter/ww8/attributeoutputbase.hxx

```diff
@@ -147,6 +147,10 @@ private:
     OUString ConvertURL( const OUString& rUrl, bool bAbsoluteOut );
 
 public:
+    OUString termUrl = "term://";
+    OUString termRefUrl = "termref://";
+    OUString sectionUrl = "section://";
+    OUString sectionRefUrl = "sectionref://";
```

Comment:
Although Sam was right that shared constants should be placed in a single place, this isn't quite how it should be done. 

1. These are defined as properties on an instance. This means that _every single instance_ allocates a new string containing these constants, instead they should be `static`. `static` means there is one instance that belongs to the class itself, not the instances of that class.
2. Also, since they are constants so define them as such. Using `constexpr` where possible, `const` if not.
3. For `OUString` instead prefer an `OUStringLiteral` to be able to use it with `constepr`
``` C++
class MyClass {
public:
static constexpr OUStringLiteral msTermUrl = u"term://";
...
```

Static methods/constants can be accessed by `ClassName::methodOrConstant`, so in this case: `MyClass::msTermUrl`.

---

However, this is also isn't a fitting place to put this. It has nothing to do with this class in which it is defined and the logic is shared between usages: checking a string if it starts with any of the prefixes.

Consider adding a header `macro_internal.hxx` to separate out our additions and put things in a namespace like this instead:

```
#pragma once

namespace macro_internal {
inline constexpr OUStringLiteral TERM_URL = u"term://";
...
}
```

Note `inline` is used to say: `this code in this C++ namespace should be compiled once for this header file`.

Since the logic is also shared, you should likely add an `isTemporaryURL` method (or some name along those lines).
`macro_internal.hxx`:
``` c++
#pragma once

namespace macro_internal {
inline constexpr OUStringLiteral TERM_URL = u"term://";
...
bool isTemporaryURL(const OUString& url);
}
```

`macro_internal.cxx`:
``` c++
#include "macro_internal.hxx"

bool macro_internal::isTemporaryURL(const OUString& url) {
using macro_internal;
return url.startsWith(TERM_URL) || ...
}
```

You would have to add the corresponding `.cxx` file in the `.mk` file

You could then use it with `macro_internal::isTemporaryURL(aDestinationURL)`